### DISCUSSION TITLE
Deselect selected option in `Select` component

### DISCUSSION THREAD

**Discussion by Abdulmajeed98** [radix-ui/primitives#2119] [Help] [answered]
Hey guys, I want to know if there is any way to de-select the already selected option when clicking again on it in a `Select` component?

**Comment by benoitgrelard** [comment, marked as answer]
Hey @Abdulmajeed98, have a look at #1569, there are a few pointers there.
Using a `key` is a good idea.

**Reply by Abdulmajeed98** [reply]
Hey @benoitgrelard, thanks for your reply.
I got one question isn't this workaround a bit expensive for react since React has to re-create everything from scratch and imagine having multiple selects in one page?

**Reply by benoitgrelard** [reply]
I wouldn't say it's expensive, but it's a workaround for now yes.

**Reply by Abdulmajeed98** [reply]
@benoitgrelard Well, I guess I have to wait until you add this feature also.
Thanks for the effort.
